### PR TITLE
[WIP] Add doc on mono-error design.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: docs/mono-error.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+= MonoError
```
Comment (akoeplinger) [inline docs/mono-error.md]: The format for headers needs to be either setext or ATX: http://daringfireball.net/projects/markdown/syntax#header

**Comment by kumpera** [comment]
@vargaz @tritao please review

File: docs/mono-error.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+MonoError is the latest attempt at cleaning up and sanitizing error handling in the runtime.
+This document highlights some of the design goals and decisions, the implementation and the migration strategy.
+
+== Design goals
+
+- Replace the majority of the adhoc error handling subsystems present today in the runtime. Each one is broken
+is a subtle way, has slightly different semantics and error conversion between them is spot, at best.
```
Comment (akoeplinger) [inline docs/mono-error.md]: ~~is~~ **in** a subtle way

File: docs/mono-error.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+- Map well to the final destination of all runtime errors: managed exceptions. This includes been compatible
```
Comment (akoeplinger) [inline docs/mono-error.md]: ~~been~~ **being** compatible

**Comment by tritao** [comment]
Looks good, my only suggestions for the future would be to try to drop the init/cleanup calls which are both error-prone and verbose. I have no suggestions how to do it in C though... If we end up moving to C++ then we could use constructors/destructors and RAII.

File: docs/mono-error.md (hunk 1 above)
Comment (kumpera) [inline docs/mono-error.md]: Done.

**Comment by vargaz** [comment]
Shouldn't this go to http://www.mono-project.com/docs/advanced/runtime/docs/ instead ?

**Comment by tritao** [comment]
@vargaz I think we should keep the docs in the Mono repo and generate the docs for the website from the files instead but if that's too much for us to deal with at the moment then putting them in the website instead sounds OK to me.

**Comment by vargaz** [comment]
We are already generating the docs from a github repo:
https://github.com/mono/website

**Comment by kumpera** [comment]
Taking Zoltan's suggestion, moving it to the website.

**Comment by kumpera** [comment]
https://github.com/mono/website/pull/119

**Pull request closed without merging**